File: OpenTool.Core/Util/StringUtil.cs
using System.Text;

namespace OpenTool.Core.Util
{
    public static class StringUtil
    {
        /// <summary>
        /// 字符串是否为空白
        /// </summary>
        /// <param name="str"></param>
        /// <returns></returns>
        public static bool IsBlank(string str)
        {
            return string.IsNullOrWhiteSpace(str);
        }

        /// <summary>
        /// 字符串是否为非空白
        /// </summary>
        /// <param name="str"></param>
        /// <returns></returns>
        public static bool IsNotBlank(string str)
        {
            return !IsBlank(str);
        }

        public static string ReplaceNonCharacters(string input, char replacement)
        {
            var sb = new StringBuilder(input.Length);
            for (var i = 0; i < input.Length; i++)
            {
                if (char.IsSurrogatePair(input, i))
                {
                    int c = char.ConvertToUtf32(input, i);
                    i++;
                    if (IsValidCodePoint(c))
                        sb.Append(char.ConvertFromUtf32(c));
                    else
                        sb.Append(replacement);
                }
                else
                {
                    char c = input[i];
                    if (IsValidCodePoint(c))
                        sb.Append(c);
                    else
                        sb.Append(replacement);
                }
            }
            return sb.ToString();
        }

        private static bool IsValidCodePoint(int point)
        {
            return point < 0xfdd0 || point >= 0xfdf0 && (point & 0xffff) != 0xffff && (point & 0xfffe) != 0xfffe && point <= 0x10ffff;
        }
    }
}

File: opentool-core/src/main/java/cn/luxio/opentool/core/util/StringUtil.java
package cn.luxio.opentool.core.util;

/**
 * 字符串工具类
 */
public class StringUtil {

    /**
     * 判断字符串是否为空白
     * <p>
     * {@code null}、空字符串以及只包含空白字符的字符串均视为空白
     *
     * @param str 待判断字符串
     * @return 为空白返回 {@code true}，否则返回 {@code false}
     */
    public static boolean isBlank(final CharSequence str) {
        final int length;

        if ((str == null) || ((length = str.length()) == 0)) {
            return true;
        }

        for (int i = 0; i < length; i++) {
            // 只要有一个非空字符即为非空字符串
            if (CharUtil.isNotBlankChar(str.charAt(i))) {
                return false;
            }
        }

        return true;
    }

    /**
     * 判断字符串是否为非空白
     *
     * @param str 待判断字符串
     * @return 非空白返回 {@code true}，否则返回 {@code false}
     */
    public static boolean isNotBlank(final CharSequence str) {
        final int length;

        if ((str == null) || ((length = str.length()) == 0)) {
            return false;
        }

        for (int i = 0; i < length; i++) {
            // 只要有一个非空字符即为非空字符串
            if (CharUtil.isNotBlankChar(str.charAt(i))) {
                return true;
            }
        }

        return false;
    }

    /**
     * 判断字符串是否为空
     * <p>
     * {@code null} 或长度为 0 的字符串视为空
     *
     * @param str 待判断字符串
     * @return 为空返回 {@code true}，否则返回 {@code false}
     */
    public static boolean isEmpty(final CharSequence str) {
        return str == null || str.isEmpty();
    }

    /**
     * 判断字符串是否为非空
     *
     * @param str 待判断字符串
     * @return 非空返回 {@code true}，否则返回 {@code false}
     */
    public static boolean isNotEmpty(final CharSequence str) {
        return !isEmpty(str);
    }
}
